File: py/1.py
def count(li, window = 1):
	result = 0
	prev = sum(li[j] for j in range(window))
	for i in range(window, len(li)):
		curr = prev + li[i] - li[i - window]
		if prev < curr:
			result += 1
		prev = curr
	return result

if __name__ == "__main__": 

	with open('../inputs/input1.txt') as f:
		lines = f.readlines()
		lines = [int(line) for line in lines]

	print('Result 1: {}'.format(count(lines, window = 1)))
	print('Result 2: {}'.format(count(lines, window = 3)))

File: js/1.js
const fs = require('fs');
const path = require("path");

const input = fs
	.readFileSync(path.resolve(__dirname, '../inputs/input1.txt'), 'utf8')
	.toString()
	.trim()
	.split('\n')
	.map((num) => parseInt(num, 10))

const count = (li, window) => {

	let result = 0;
	let prev = 0;

	for (let j = 0; j < window; j += 1) {
		prev += li[j];
	}

	for (let i = window; i < li.length; i += 1) {
		const curr = prev + li[i] - li[i - window]
		if (prev < curr) {
			result += 1
		}
		prev = curr
	}

	return result;
}


const res1 = count(input, 1)
const res2 = count(input, 3)

console.log(`Result 1: ${res1}`)
console.log(`Result 2: ${res2}`)